common/ferrostar/src/routing_adapters/osrm/algorithms.rs

```diff
@@ -0,0 +1,83 @@
+use super::models::{Annotation, AnnotationValue, MaxSpeed, MaxSpeedUnits};
+use itertools::izip;
+
+/// Get's he slice of annotations
+pub fn get_annotation_slice(
+    start_index: usize,
+    end_index: usize,
+    annotations: Option<Vec<AnnotationValue>>,
+) -> Option<Vec<AnnotationValue>> {
+    return if let Some(annotations) = &annotations {
+        let annot_len = annotations.len();
+
+        println!("Step Indexes: (start_index = {start_index}, end_index={end_index}, annotation.len: {annot_len})");
+
+        let slice = &annotations[start_index..end_index];
+        Some(slice.to_vec())
+    } else {
+        None
+    };
+}
+
+/// Converts all single value annotation vectors into a single vector of values.
+pub fn zip_annotations(annotations: Annotation) -> Vec<AnnotationValue> {
+    izip!(
+        annotations.distance,
+        annotations.duration,
+        annotations.max_speed,
+        annotations.speed
+    )
+    .map(|(distance, duration, max_speed, speed)| AnnotationValue {
+        distance,
+        duration,
+        max_speed_mps: convert_max_speed_to_mps(max_speed),
+        speed_mps: speed,
+    })
+    .collect()
+}
+
+/// Converts a max speed value to meters per second.
+pub fn convert_max_speed_to_mps(max_speed: MaxSpeed) -> Option<f64> {
+    match max_speed {
+        MaxSpeed::Known { speed, unit } => match unit {
+            MaxSpeedUnits::KilometersPerHour => return Some(speed * 0.27778),
+            MaxSpeedUnits::MilesPerHour => return Some(speed * 0.44704),
+        },
+        #[allow(unused)]
+        MaxSpeed::Unknown { unknown } => return None,
+    }
+}
```

Comment:
I think this can be exposed as an implementation method on the `MaxSpeed` enum itself.

Reply:
This function gone 👍